sm_ipt/doc/usage.rst

```diff
@@ -0,0 +1,228 @@
+
+.. code-block:: c
+
+	struct sm_ipt_ctx sm_ipt_ctx;
+
+	void sm_ipt_receive_handler(const uint8_t *packet, size_t len)
+	{
+		/* Process data */
+		some_data_processing_function();
+	}
+
+	int initialize(void)
+	{
+		return sm_ipt_init(&sm_ipt_ctx, sm_ipt_receive_handler);
+	}
+
+The above code uses the :c:func:`sm_ipt_init` function to initialize the :c:struct:`sm_ipt_ctx` SM IPT context data structure and to assign the receive handler.
+
+Data send full-copy
+-------------------
+
+The data is prepared beforehand, copied into the allocated buffer and sent to another core.
+
+.. code-block:: c
+
+	int send(const void *data, size_t len)
+	{
+		uint8_t *buffer;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		memcpy(buffer, data, len);
+
+		return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+	}
+
+The above code uses the :c:func:`sm_ipt_alloc_tx_buf` function to allocate the transmit buffer in the shared memory.
+The data is copied into the said buffer and sent to another core through the :c:func:`sm_ipt_send` function.
+
+Data send no-copy
+-----------------
+
+It is possible to operate directly on the transmit buffer to avoid copying from the local buffer.
+
+.. code-block:: c
+
+	int send_nocopy(size_t len)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		/* Use buffer as any other allocated buffer */
+		some_data_preparation_function(buffer);
+
+		return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+	}
+
+The above code uses the :c:func:`sm_ipt_alloc_tx_buf` function to allocate the transmit buffer in the shared memory.
+The buffer is prepared by the :c:func:`some_data_preparation_function` and sent to another core through the :c:func:`sm_ipt_send` function.
+
+Data send abort
+---------------
+
+It is possible to free the transmit buffer instead of sending it.
+
+.. code-block:: c
+
+	int send_if(size_t len, bool send_choice)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		if (send_choice){
+			/* Prepare some data */
+			some_data_preparation_function(buffer);
+
+			return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+		}
+		else{
+			sm_ipt_free_tx_buf(&sm_ipt_ctx, buffer);
+
+			return ERROR_ABORT;
+		}
+	}
+
+The above code uses the :c:func:`sm_ipt_alloc_tx_buf` function to allocate the transmit buffer in the shared memory.
+The buffer is sent only if ``send_choice`` is true. If ``send_choice`` is false the buffer is free using the :c:func:`sm_ipt_free_tx_buf` and the send operation is aborted.
+
+Data receive full-copy
+----------------------
+
+When data is received, it can be processed using the full-copy mechanism.
+The buffer must be copied to the local memory.
+The shared memory rx buffer must be freed and data processing takes place on the local copy of the buffer.
+
+.. code-block:: c
+
+	volatile uint8_t receive_buffer[MAX_PACKET_SIZE];
+	volatile bool data_received;
```

Comment:
I am not sure that using volatiles int this sample is a good idea. Maybe use some generic functions names, e.g. 

```c
/* Write content of the buffer to some pipe. */
write_bytes_to_pipe(packet, len);

/* Free receive buffer */
sm_ipt_free_rx_buf(&sm_ipt_ctx, packet);
```

And then, `data_processing_thread` is not needed.